source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -15,7 +15,32 @@ namespace ConnPool {
 
 ConfigImpl::ConfigImpl(
     const envoy::config::filter::network::redis_proxy::v2::RedisProxy::ConnPoolSettings& config)
-    : op_timeout_(PROTOBUF_GET_MS_REQUIRED(config, op_timeout)) {}
+    : op_timeout_(PROTOBUF_GET_MS_REQUIRED(config, op_timeout)),
+      hashtagger_(makeHashtagger(config.hashtags())) {}
+
+// Inspired by the redis-cluster hashtagging algorithm
+// https://redis.io/topics/cluster-spec#keys-hash-tags
+std::string ConfigImpl::hashtag(const std::string& s) {
```

Comment:
does it make sense to have this as a member of ConfigImpl? i think it would be better to include it on the LbContextImpl.

Reply:
I was considering this too, WDYT of moving the hashtagging function into LbContextImpl, then invoking it lazily?